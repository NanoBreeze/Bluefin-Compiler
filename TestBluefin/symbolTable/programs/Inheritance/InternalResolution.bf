/*
We expect a derived class to be able to use its base class' members and methods
*/

struct Base {
	int a;
	void f() {}
};

struct Der extends Base {
	void g() {
		a = 5;
		f();
		b = 1;
	}
};
