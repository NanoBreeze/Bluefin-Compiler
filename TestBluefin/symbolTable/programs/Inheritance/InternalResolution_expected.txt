declare - Base - c_struct - t_Base
setCurrentScope - Level 1
declare - a - c_var - t_int
declare - f - c_function - t_void
enterScope - Level 2
exitScope - Level 2
exitScope - Level 1
resolve - Base - c_struct - t_Base
declare - Der - c_struct - t_Der
setCurrentScope - Level 1
declare - g - c_function - t_void
enterScope - Level 2
resolve - Base::g - UNRESOLVED
exitScope - Level 2
exitScope - Level 1
resolve - int - c_builtinType - t_int
resolve - void - c_builtinType - t_void
resolve - void - c_builtinType - t_void
resolve - a - c_var - t_int
resolve - f - c_function - t_void
resolve - b - UNRESOLVED
